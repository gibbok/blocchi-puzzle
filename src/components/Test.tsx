import React from 'react';

export function Test(): JSX.Element {
  const [counter, setCounter] = React.useState(0);

  React.useEffect(() => {
    const interval = setInterval(() => {
      setCounter(counter + 1);
    }, 1000);

    return () => clearInterval(interval);
  });

  return <div>{counter}</div>;
}
