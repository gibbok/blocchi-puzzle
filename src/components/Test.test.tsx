import React from 'react';
import renderer, { act } from 'react-test-renderer';
import { Test } from './Test';

describe('test', () => {
  jest.useFakeTimers();
  it('should xx', () => {
    act(() => {
      const tree = renderer.create(<Test />);

      jest.advanceTimersByTime(0);
      expect(tree).toMatchSnapshot();

      jest.advanceTimersByTime(5000);
      expect(tree).toMatchSnapshot();

      jest.advanceTimersByTime(10000);
      expect(tree).toMatchSnapshot();
    });
  });
});
