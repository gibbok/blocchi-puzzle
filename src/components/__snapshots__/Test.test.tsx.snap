// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test should xx 1`] = `
<div>
  5
</div>
`;

exports[`test should xx 2`] = `
<div>
  15
</div>
`;
